### DISCUSSION TITLE
Is it a bug report or security vulnerability?

### DISCUSSION THREAD

**Discussion by ikudrickiy** [excaliburjs/Excalibur#3007] [Q&A]
The question may seem stupid. But I'm not afraid of those 😉

[This public function](https://github.com/excaliburjs/Excalibur/blob/c0b6cdf7c9439f840f76d3c81bc4b991ea19c357/src/engine/Collision/BoundingBox.ts#L259) returns the same array referrence for multiple function calls. I can change the array in one place and it will be changed for all the calls. At the same time, if we assume cloning I don't understand the need for a caching mechanism. It looks like a burden in this case. And if there's no caching, cloning seems like a burden.

We can solve this problem on a typescript level turning the function return into readonly array ([see this](https://stackoverflow.com/a/57923217/20427555)).

**Comment by eonarheim** [comment]
@ikudrickiy I think a readonly array is appropriate here. The intent is to cache the points of the bounding box to avoid unnecessary JS allocations in the hot path

**Reply by eonarheim** [reply]
@ikudrickiy I've made a bug for this! https://github.com/excaliburjs/Excalibur/issues/3020
